### PR TITLE
Add `first` and `coalesce` resampling functions

### PR DESCRIPTION
**Pull request by phillip-wenig-frequenz**
- `first` takes the first value in an interval, regardless of being None.
- `coalesce` takes the first non-none value in an interval. If all are None, returns None.

### CODE REVIEW

File: tests/test_resampler.py [hunk 1]
```diff
@@ -223,6 +277,10 @@ def test_resampling_function_name_value() -> None:
     assert ResamplingFunction.Last.value == 4
     assert ResamplingFunction.Count.name == "Count"
     assert ResamplingFunction.Count.value == 5
+    assert ResamplingFunction.First.name == "First"
+    assert ResamplingFunction.First.value == 6
+    assert ResamplingFunction.Coalesce.name == "Coalesce"
+    assert ResamplingFunction.Coalesce.value == 7
```
Comment (cwasicki) [inline tests/test_resampler.py:295]: Is this test even needed, i.e. does something rely on the mapping between names and values or the name strings (e.g. the python code)?

File: tests/test_resampler.py [hunk 2]
```diff
@@ -128,6 +152,30 @@ def test_resampler_resampling_function_last() -> None:
...
+        max_age_in_intervals=1,
+        start=start,
+    )
+
+    for i in range(10):
+        resampler.push_sample(timestamp=start + i * step, value=i + 1)
+
+    expected = [
+        (start + 5 * step, 1.0),
+        (start + 10 * step, 6.0),
```
Comment (cwasicki) [inline tests/test_resampler.py]: I suggest to have some Nones in the test case

**Review by cwasicki — DISMISSED** [review]
LGTM, some minor comments.

File: tests/test_resampler.py (hunk 1 above)
Comment (phillip-wenig-frequenz) [reply tests/test_resampler.py:295]: Unfortunately, an Enum in Rust is not an Enum in Python after creating the bindings. Therefore, we have to rebuild it in a way to have the same behavior an Enum would have. This is tested here.

File: tests/test_resampler.py (hunk 2 above)
Comment (phillip-wenig-frequenz) [reply tests/test_resampler.py]: There are already some in the Rust tests, but I will add some here as well.

**Comment by llucax** [comment]
Do we have use cases for these?

**Review by cwasicki — APPROVED** [review]

**Pull request merged**